From 4250332f461aab7fdb9184c35d305010b7cbf8a7 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Thu, 26 Jan 2023 04:22:34 -0800
Subject: [PATCH] Update Makefile for OE compatibility

* Use install instead of cp
* Use GBS standard nams for target directory variables

Signed-off-by: Matt Madison <matt@madison.systems>
---
 optee/optee_client/Makefile | 40 +++++++++++++++++++------------------
 1 file changed, 21 insertions(+), 19 deletions(-)

diff --git a/optee/optee_client/Makefile b/optee/optee_client/Makefile
index 2088cc9..6cafe72 100644
--- a/optee/optee_client/Makefile
+++ b/optee/optee_client/Makefile
@@ -12,9 +12,9 @@ export VPREFIX
 
 EXPORT_DIR ?= $(O)/export
 DESTDIR ?= $(EXPORT_DIR)
-SBINDIR ?= /usr/sbin
-LIBDIR ?= /usr/lib
-INCLUDEDIR ?= /usr/include
+sbindir ?= /usr/sbin
+libdir ?= /usr/lib
+includedir ?= /usr/include
 
 .PHONY: all build build-libteec build-libckteec build-libseteec \
 	build-libteeacl check-libuuid install copy_export clean cscope \
@@ -152,19 +152,21 @@ checkpatch-all-files: checkpatch-pre-req
 distclean: clean
 
 copy_export: build
-	mkdir -p $(DESTDIR)$(SBINDIR) $(DESTDIR)$(LIBDIR) $(DESTDIR)$(INCLUDEDIR)
-	cp config.mk $(DESTDIR)/$(INCLUDEDIR)/optee_client_config.mk
-	cp -d ${O}/libteec/libteec.so* $(DESTDIR)$(LIBDIR)
-	cp -d ${O}/libteec/libteec.a $(DESTDIR)$(LIBDIR)
-	cp ${O}/tee-supplicant/tee-supplicant $(DESTDIR)$(SBINDIR)
-	cp public/*.h $(DESTDIR)$(INCLUDEDIR)
-	cp libckteec/include/*.h $(DESTDIR)$(INCLUDEDIR)
-	cp -d ${O}/libckteec/libckteec.so* $(DESTDIR)$(LIBDIR)
-	cp -d ${O}/libckteec/libckteec.a $(DESTDIR)$(LIBDIR)
-# Disable libteeacl:
-#	cp libteeacl/include/*.h $(DESTDIR)$(INCLUDEDIR)
-#	cp -d ${O}/libteeacl/libteeacl.so* $(DESTDIR)$(LIBDIR)
-#	cp -d ${O}/libteeacl/libteeacl.a $(DESTDIR)$(LIBDIR)
-	cp libseteec/include/*.h $(DESTDIR)$(INCLUDEDIR)
-	cp -d ${O}/libseteec/libseteec.so* $(DESTDIR)$(LIBDIR)
-	cp -d ${O}/libseteec/libseteec.a $(DESTDIR)$(LIBDIR)
+	install -d $(DESTDIR)$(sbindir) $(DESTDIR)$(libdir) $(DESTDIR)$(includedir)
+	install -m 0644 config.mk $(DESTDIR)/$(includedir)/optee_client_config.mk
+	install -m 0644 ${O}/libteec/libteec.so.1.0.0 $(DESTDIR)$(libdir)
+	ln -sf libteec.so.1.0.0 $(DESTDIR)$(libdir)/libteec.so.1.0
+	ln -sf libteec.so.1.0.0 $(DESTDIR)$(libdir)/libteec.so.1
+	install -m 0644 ${O}/libteec/libteec.a $(DESTDIR)$(libdir)
+	install -m 0755 ${O}/tee-supplicant/tee-supplicant $(DESTDIR)$(sbindir)
+	install -m 0644 public/*.h $(DESTDIR)$(includedir)
+	install -m 0644 libckteec/include/*.h $(DESTDIR)$(includedir)
+	install -m 0644 ${O}/libckteec/libckteec.so.0.1.0 $(DESTDIR)$(libdir)
+	ln -sf libckteec.so.0.1.0 $(DESTDIR)$(libdir)/libckteec.so.0.1
+	ln -sf libckteec.so.0.1.0 $(DESTDIR)$(libdir)/libckteec.so.0
+	install -m 0644 ${O}/libckteec/libckteec.a $(DESTDIR)$(libdir)
+	install -m 0644 libseteec/include/*.h $(DESTDIR)$(includedir)
+	install -m 0644 ${O}/libseteec/libseteec.so.0.1.0 $(DESTDIR)$(libdir)
+	ln -sf libseteec.so.0.1.0 $(DESTDIR)$(libdir)/libseteec.so.0.1
+	ln -sf libseteec.so.0.1.0 $(DESTDIR)$(libdir)/libseteec.so.0
+	install -m 0644 ${O}/libseteec/libseteec.a $(DESTDIR)$(libdir)
-- 
2.34.1

